src/p2f/src/CodeOwls.PowerShell/CodeOwls.PowerShell.Provider/Provider.cs

```diff
@@ -534,7 +567,7 @@ private string DoMakePath(string parent, string child)
             //trim the end slash for the consistent experience to other built-in providers such as FileSystem.
             //Show: drive:\A\B\C>
             //Not:  drive:\A\B\C\>
-            var newChild = child.TrimEnd('\\');
+            var newChild = child.TrimEnd('/', '\\');
```

Comment:
Path.GetDirectoryName should do this for you.

Reply:
We cannot replace it with Path.GetDirectoryName because some path may not have end slash.